scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/SchemaProvider.scala

```diff
@@ -81,9 +83,36 @@ private[types] object SchemaProvider {
       case _ => ("REQUIRED", tpe)
     }
     val (tpeParam, nestedParam) = rawType(valType)
-    field(mode, name, tpeParam, nestedParam)
+    field(mode, name, tpeParam, desc, nestedParam)
   }
 
   private def toFields(t: Type): Iterable[TableFieldSchema] = getFields(t).map(toField)
 
+  private def getFields(t: Type): Iterable[(Symbol, Option[String])] =
+    t.declarations.filter(isField) zip fieldDescs(t)
+
+  private def fieldDescs(t: Type): Iterable[Option[String]] = {
+    val tpe = "com.spotify.scio.bigquery.types.description"
+    // TODO: scala 2.11
+    /*
+    t.typeSymbol.asClass.primaryConstructor.typeSignature.paramLists.head.map {
+      _.annotations
+        .find(_.tree.tpe.toString == tpe)
+        .map { a =>
+          val q"new $t($v)" = a.tree
+          val Literal(Constant(s)) = v
+          s.toString
+        }
+    }
+    */
+    t.declaration(nme.CONSTRUCTOR).asMethod.paramss.head.map {
+      _.annotations
+        .find(_.tpe.toString == tpe)
+        .map { a =>
```

Comment:
That'll make a copy of the list though, not a big deal but no reason to fuss around here?